ironfish/src/memPool/memPool.ts

```diff
@@ -42,7 +42,12 @@ export class MemPool {
   /**
    * Accepts a transaction from the network
    */
-  acceptTransaction(transaction: Transaction): boolean {
+  async acceptTransaction(transaction: Transaction): Promise<boolean> {
+    const { valid } = await this.chain.verifier.verifyTransaction(transaction)
+    if (!valid) {
```

Comment:
We should add a log here about the invalid transaction and the reason

Reply:
We should add ban score for peers here later, I just haven't thought through all the details yet.